homeassistant/helpers/template.py

```diff
@@ -949,6 +953,87 @@ def is_device_attr(
     return bool(device_attr(hass, device_or_entity_id, attr_name) == attr_value)
 
 
+def area_id(hass: HomeAssistant, lookup_value: str) -> str | None:
+    """Get the area ID from an area name, device id, or entity id."""
+    area_reg = area_registry.async_get(hass)
+    area = area_reg.async_get_area_by_name(str(lookup_value))
+
+    try:
```

Comment:
If we already have a valid area from the above, we could save ourselves a little by not trying to resolve the rest?

Reply:
I was trying to be clever and handle use cases where an area name could match a device ID or entity ID, but I suppose it's not worth trying to handle those use cases?

Reply:
I think one has a better shot in buying a lottery ticket 😬